component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAuthenticator.java

```diff
@@ -643,7 +609,7 @@ private void removeUsedBackupCode(String userToken, String username, List<String
             UserRealm userRealm = BackupCodeUtil.getUserRealm(username);
             UserStoreManager userStoreManager = userRealm.getUserStoreManager();
             Map<String, String> claimsToUpdate = new HashMap<>();
-            claimsToUpdate.put(BACKUP_CODES_CLAIM, encryptedBackupCodes);
+            claimsToUpdate.put(BACKUP_CODES_CLAIM, unusedBackupCodes);
             userStoreManager.setUserClaimValues(tenantAwareUsername, claimsToUpdate, null);
```

Comment:
We could have written a new method called update user and remove this logic being repeated. WDYT?